### PR TITLE
Collect more metrics in publisher & consumer

### PR DESCRIPTION
**Pull request by tayloraburgess**

### CODE REVIEW

**Comment by mattbriancon** [comment]
What other metrics would you want to have if you had to evaluate the health of the messaging system? We can chat IRL if that'd be easier.

**Comment by tayloraburgess** [comment]
@mattbriancon Made some updates here as per our conversation earlier, if you want to take another look.

File: mbq/pubsub/consumer.py [hunk 1]
```diff
@@ -115,17 +132,21 @@ def process_queue(self):
...
 
+            process_time_ms = utils.timedelta_to_ms(arrow.now() - start_process)
             collector.increment(
                 "consumer.processed",
-                tags={"result": result, "message_type": message_type, "queue": self._queue_name},
+                tags={
+                    "result": result,
+                    "message_type": message_type,
+                    "queue": self._queue_name,
+                    "process_time_ms": process_time_ms,
```
Comment (mattbriancon) [inline mbq/pubsub/consumer.py]: I'm not sure we'll be able to do anything useful with the timing if it's in a tag. We could instead fire a `timing` metric like `consumer.process_duration_ms` with the timing as the value and then keep the rest of the tags.

File: mbq/pubsub/utils.py [hunk 2]
```diff
@@ -32,38 +33,50 @@ def __init__(
...
             "message_type": self.message_type,
             "payload": self.payload,
             "payload_type": self.payload_type.value,
-            "envelope_created_at": self.envelope_created_at.format(),
+            "envelope_created_at": self.envelope_created_at.format()
+            if self.envelope_created_at
+            else None,
+            "dlq_envelope_created_at": self.dlq_envelope_created_at.format()
+            if self.dlq_envelope_created_at
+            else None,
```
Comment (mattbriancon) [inline mbq/pubsub/utils.py]: Oof these ternaries are brutal 😅

Comment (tayloraburgess) [reply mbq/pubsub/utils.py]: Yeah, not a fan of the way black formats theese

File: mbq/pubsub/consumer.py (hunk 1 above)
Comment (tayloraburgess) [reply mbq/pubsub/consumer.py]: Ah, good call

File: mbq/pubsub/consumer.py [hunk 3]
```diff
@@ -89,14 +86,40 @@ def process_queue(self):
...
+                    "consumer.read.from_envelope_creation_ms",
+                    utils.timedelta_to_ms(now - envelope.envelope_created_at),
+                    tags=tags,
+                )
+            try:
+                body = json.loads(message.body)
+                if "Timestamp" in body:
+                    collector.timing(
+                        "consumer.read.from_message_creation_ms",
+                        utils.timedelta_to_ms(now - arrow.get(body["Timestamp"])),
```
Comment (mattbriancon) [inline mbq/pubsub/consumer.py]: Remind me which timestamp `body["Timestamp"]` is?

**Pull request merged**